src/iuse_actor.cpp

```diff
@@ -4351,14 +4358,17 @@ cata::optional<int> detach_gunmods_actor::use( Character &p, item &it, bool,
 ret_val<bool> detach_gunmods_actor::can_use( const Character &p, const item &it, bool,
         const tripoint & ) const
 {
-    const auto mods = it.gunmods();
+    const std::vector<const item *> mods = it.gunmods();
```

Comment:
Is this really a vector of pointers? It seems to me that we should use references here. A nullpointer in the list makes no sense to me but I'm not really familiar with that code.

Reply:
Okay, maybe that shouldn't be done in this PR but I was thinking about e.g. option 2 [here](https://stackoverflow.com/questions/23488326/stdvector-of-references), i.e. reference_wrapper. I think the overhead regarding performance and code complexity should be negligible.

Reply:
That's definitely out of scope for this PR.